dandi/files/zarr.py

```diff
@@ -56,31 +57,122 @@
 lgr = get_logger()
 
 
-def get_zarr_format_version(data: Any) -> str:
+class ZarrValidationError(Exception):
     """
-    Get the Zarr storage specification version from a Zarr data object
+    Exception raised when handling a dataset to be validated against the Zarr standard
+    is deemed impossible to be valid
+    """
+
+
+class _Zarr3Metadata(BaseModel):
+    """
+    Metadata for Zarr format V3 stored in the zarr.json file
+
+    Note
+    ----
+        This will not be needed once the upgrade to zarr-python 3.x is done and
+        should be removed.
+    """
+
+    node_type: str
+
+
+def get_zarr_format_version(ds_path: Path) -> str:
```

Comment:
keep variable as just path to avoid confusion with datasets / dandisets.
```suggestion
def get_zarr_format_version(path: Path) -> str:
```

Reply:
done